pkg/netdevice/pciNetDevice.go

```diff
@@ -31,35 +31,46 @@ type pciNetDevice struct {
 	linkSpeed string
 	rdmaSpec  types.RdmaSpec
 	linkType  string
+	vdpaDev   types.VdpaDevice
 }
 
 // NewPciNetDevice returns an instance of PciNetDevice interface
 func NewPciNetDevice(dev *ghw.PCIDevice, rFactory types.ResourceFactory, rc *types.ResourceConfig) (types.PciNetDevice, error) {
 	var ifName string
 	infoProviders := make([]types.DeviceInfoProvider, 0)
+	var vdpaDev types.VdpaDevice
 
 	driverName, err := utils.GetDriverName(dev.Address)
 	if err != nil {
 		return nil, err
 	}
 
-	infoProviders = append(infoProviders, rFactory.GetDefaultInfoProvider(dev.Address, driverName))
 	rdmaSpec := rFactory.GetRdmaSpec(dev.Address)
 	nf, ok := rc.SelectorObj.(*types.NetDeviceSelectors)
 	if ok {
 		// Add InfoProviders based on Selector data
-		if nf.IsRdma {
-			if rdmaSpec.IsRdma() {
-				infoProviders = append(infoProviders, NewRdmaInfoProvider(rdmaSpec))
-			} else {
-				glog.Warningf("RDMA resources for %s not found. Are RDMA modules loaded?", dev.Address)
+		if nf.VdpaType != "" {
+			vdpaDev = rFactory.GetVdpaDevice(dev.Address)
+			if vdpaDev == nil {
+				glog.Warningf("No vDPA device found for device %s", dev.Address)
 			}
-		}
-		if nf.NeedVhostNet {
-			if VhostNetDeviceExist() {
-				infoProviders = append(infoProviders, NewVhostNetInfoProvider())
-			} else {
-				glog.Errorf("GetDeviceSpecs(): vhost-net is required in the configuration but /dev/vhost-net doesn't exist")
+			infoProviders = append(infoProviders, NewVdpaInfoProvider(nf.VdpaType, vdpaDev))
```

Comment:
I have previously commented on it.

IMHO i think the `genericInfoProvider` should still be present to provide the PCI address.

Reply:
Agreed. Will change it.

Reply:
Done